digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/FloatingChecker.scala

```diff
@@ -60,31 +65,50 @@ class FloatingChecker(roadLinkService: RoadLinkService) {
   }
 
   /**
-    * Check if road address geometry is moved by road link geometry change at leas MaxMoveDistanceBeforeFloating
+    * Check if road address geometry is moved by road link geometry change at least MaxMoveDistanceBeforeFloating
     * meters. Because road address geometry isn't directed check for fit either way. Public for testing.
     * @param roadLink Road link for road address list
     * @param roadAddresses Sequence of road addresses to check
     * @return true, if geometry has changed for any of the addresses beyond tolerance
     */
   def isGeometryChange(roadLink : RoadLinkLike, roadAddresses: Seq[RoadAddress]) : Boolean = {
-    roadAddresses.exists(ra => {
+    val movedAddresses = roadAddresses.filter(ra => {
       GeometryUtils.geometryMoved(MaxMoveDistanceBeforeFloating)(
-        GeometryUtils.truncateGeometry3D(roadLink.geometry, ra.startMValue, ra.endMValue),
+        GeometryUtils.truncateGeometry2D(roadLink.geometry, ra.startMValue, ra.endMValue),
         ra.geometry) &&
         GeometryUtils.geometryMoved(MaxMoveDistanceBeforeFloating)(
-          GeometryUtils.truncateGeometry3D(roadLink.geometry, ra.startMValue, ra.endMValue),
+          GeometryUtils.truncateGeometry2D(roadLink.geometry, ra.startMValue, ra.endMValue),
           ra.geometry.reverse) // Road Address geometry isn't necessarily directed: start and end may not be aligned by side code
     }
-    ) || Math.abs(roadAddresses.maxBy(_.endMValue).endMValue - GeometryUtils.geometryLength(roadLink.geometry)) > MaxMoveDistanceBeforeFloating
+    )
+    if(movedAddresses.size != 0) {
```

Comment:
Use Scala functionalities in this kind of validations. moveAddresses.isNonEmpty could do the work, or even better, avoiding the "if else" logic and use "match" to make the code more readable